core/includes/theme.inc

```diff
@@ -1951,7 +1951,19 @@ function theme_dropbutton_wrapper($variables) {
  */
 function theme_image($variables) {
   $attributes = $variables['attributes'];
-  $attributes['src'] = file_create_url($variables['uri']);
+
+  // Check for missing 'uri', allow D7's 'path' instead.
+  $uri = $variables['uri'];
+  if (!$uri && !empty($variables['path'])) {
```

Comment:
Excuse newbie comment, but isn't there a chance some people (for some "exotic" reason like trying to keep their code compatible with both D7 and Backdrop) might use both `uri` and `path`?. If so, shouldn't we be checking only `if (!empty($variables['path']))` to cover for these cases? Alternatively, we can do this:

```php
// 'path' is defined.
if (!empty($variables['path'])) {
  $message = 'theme_image() called with deprecated "path", use "uri" instead.';
  // 'uri' is also defined.
  if ($uri) {
    $message = 'Deprecated "path" used in theme_image(), use of "uri" is sufficient.';
  }
  // 'uri' not defined; set it to the same value as 'path'.
  else {
    $uri = $variables['path'];
  }
  watchdog('image', $message, NULL, WATCHDOG_DEPRECATED, 'https://api.backdropcms.org/api/backdrop/core%21includes%21theme.inc/function/theme_image/1');
}
$attributes['src'] = file_create_url($uri);
```

Perhaps I'm over-complicating things though.